### PR TITLE
User model and authentication

### PR DESCRIPTION
**Pull request by jjwyse**

### CODE REVIEW

File: app/controllers/authentication_controller.rb [hunk 1]
```diff
@@ -1,8 +1,9 @@
 class AuthenticationController < ApplicationController
   def authenticate
-    email = params[:email]
-    password = params[:password]
-    puts email, password
-    render json: {success: true}, status: 200
+    user = User.find_by(email: params[:email])
+    isAuthenticated = !!user.authenticate(params[:password])
+    status = isAuthenticated ? 200 : 401
+    puts params[:email], params[:password], user, isAuthenticated, status;
```
Comment (jjwyse) [inline app/controllers/authentication_controller.rb]: ⚔️

File: app/controllers/authentication_controller.rb [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
+    render json: {success: isAuthenticated}, status: status
```
Comment (jjwyse) [inline app/controllers/authentication_controller.rb]: ```status: isAuthenticated ? 200 : 401```

File: app/models/user.rb [hunk 3]
```diff
@@ -1,4 +1,12 @@
 class User < ApplicationRecord
-  has_secure_token :auth_token
+  before_save { self.email = email.downcase }
+  validates :first_name,  presence: true, length: { maximum: 50 }
+  validates :last_name,  presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
```
Comment (jjwyse) [inline app/models/user.rb]: Make this a constant.  I think you do `.freeze` in ruby but may want to google that ...

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
```
Comment (jjwyse) [inline spec/models/user_spec.rb]: We're not using Rspec for tests.  So get rid of rspec from this and then make some tests?

File: test/fixtures/users.yml [hunk 5]
```diff
@@ -1,18 +1,2 @@
 # TODO - JJW - read about these 👇
 # Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
-
-one:
```
Comment (jjwyse) [inline test/fixtures/users.yml]: Why'd you kill these?

File: spec/models/user_spec.rb (hunk 4 above)
Comment (jjwyse) [reply spec/models/user_spec.rb]: Oh, just noticed there is a `user_test` down below.  So maybe just remove this?

Comment (laurawyse) [reply spec/models/user_spec.rb]: yeah, this was autogenerated by something but I can remove it.

**Comment by jjwyse** [comment]
Yeah, it's been happening to me too.  Need to figure out how to either turn
of the rspec generator pieces or adopt rspec for testing going forward ...

File: spec/models/user_spec.rb (hunk 4 above)
Comment (jjwyse) [reply spec/models/user_spec.rb]: Yeah, it's been happening to me too.  Need to figure out how to either turn
of the rspec generator pieces or adopt rspec for testing going forward ...

**Pull request merged**